data_structures/binary-search-tree/bst.py

```diff
@@ -0,0 +1,110 @@
+class Node:
+    """Create node class."""
+    def __init__(self, val):
+        self.val = val
+        self.right = None
+        self.left = None
+
+    def __repr__(self):
+        return '<Node Val: {}'.format(self.val)
+
+    def __str__(self):
+        return self.val
+
+
+class BST:
+    """Create binary search tree class."""
+    def __init__(self, iter=[]):
+        self.root = None
+
+        try:
+            for item in iter:
+                self.insert(item)
+        except ValueError:
+            print('Value error! Please insert an iterable.')
+
+    def __repr__(self):
+        if self.root:
+            return '<BST Root {}>'.format(self.root.val)
+        else:
+            raise AttributeError('Tree is empty!')
```

Comment:
This should probably not raise an exception because this should always show the user the current state of the tree. Thus you could return something like `'<BST Root None>'` to show that the tree is currently empty.